### PR TITLE
Add lane that deploys Test Flight

### LINKED ISSUE #14: Include lane to upload to iTunes Connect
A crucial part of our workflow not yet covered by the MasterFastfile

### PR DESCRIPTION
**Pull request by NikolaKirev**
This PR closes #14

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -19,6 +19,9 @@ FL_HOCKEY_TEAMS="" #Hockey ID (number not name)
 FL_HOCKEY_NOTIFY= #Email team when new build avialable? 0 = No, 1 = Yes
 ICON_OVERLAY_ASSETS_BUNDLE="" #Path to .xcassets
 TAB_USE_TIME_FOR_BUILD_NUMBER= #Use Time and date for build number or BUILD_NUMBER environment variable (created by jenkis or team city) true = use time, false = use BUILD_NUMBER
+ITUNES_CONNECT_USERNAME="" #iTunes Connect login (usually email address)
+ITUNES_CONNECT_TEAM_ID="" #The ID of your iTunes Connect team if you're in multiple teams
```
Comment (KaneCheshire) [inline setup.sh:23]: Can we add this link somewhere on how to find this value? https://github.com/fastlane/fastlane/issues/4301#issuecomment-253461017

Comment (NikolaKirev) [reply setup.sh:23]: Done. Added in README where the lane is documented.

**Review by jsanderson44 — APPROVED** [review]

File: setup.sh (hunk 1 above)
Comment (KaneCheshire) [reply setup.sh:23]: Nice one :)

File: Fastfile [hunk 2]
```diff
@@ -26,6 +26,15 @@ default_platform :ios
     _buildAndDeployToHockey()
   end
 
+  lane :deploy_to_test_flight do
+    _setup()
+    scan
+    icon_overlay(version: get_version_number)
+    _set_build_number()
+	  _build_ipa()
```
Comment (KaneCheshire) [inline Fastfile]: Slight indentation issue if I'm being pedantic :P

**Review by KaneCheshire — APPROVED** [review]

File: Fastfile [hunk 3]
```diff
@@ -26,6 +26,15 @@ default_platform :ios
     _buildAndDeployToHockey()
   end
 
+  lane :deploy_to_test_flight do
+    _setup()
+    scan
+    icon_overlay(version: get_version_number)
```
Comment (Neil3079) [inline Fastfile]: we shouldn't do this for release builds

File: README.md [hunk 4]
```diff
@@ -37,6 +37,12 @@ For more detailed instructions [see our wiki](https://github.com/theappbusiness/
   * builds and archives project
   * generates changelog from git commits
   * uploads app to hockey
+* `deploy_to_test_flight` (How do I find my iTunes Connect team ID? [link](https://github.com/fastlane/fastlane/issues/4301#issuecomment-253461017))
+  * runs tests
+  * sets build number
+  * adds build info to app icon
```
Comment (Neil3079) [inline README.md:43]: again this needs to be removed as we don't want build info on our app store app icon

File: Fastfile (hunk 3 above)
Comment (NikolaKirev) [reply Fastfile]: Good point. Should we have a separate `deploy_to_app_store` lane on get rid of the icon overlay altogether when submitting to iTC?

**Review by Neil3079 — APPROVED** [review]

**Pull request merged**